lms/templates/oauth2_provider/README.rst

```diff
@@ -0,0 +1,16 @@
+Django Oauth Toolkit Templates
+------------------------------
+
+In an ideal world, these templates would live in the ``oauth_dispatch`` djangoapp along
+with our other oauth2 code. Unfortunately, this template is defined within the
+``django-oauth-toolkit`` library, which also defines its own version of the template.
+
+Due to `the way that Django's app_directories loader works`_, we cannot put this template
+in ``oauth_dispatch``, becuase the `template defined in the library`_ will take precendence
+over the one in ``oauth_dispatch``. The library must be defined first in ``INSTALLED_APPS``.
```

Comment:
Changes in this suggestion:
* Updated intro to explain the purpose of the directory before detailing why it needs to be this way.
* Adjusted the following paragraph to work with the above change.
* A couple of type fixes.
* Change "the library" to "django-oauth-toolkit" in the link.

```suggestion
This Django app exists solely to provide a home for the ``authorize.html`` template. This overrides the default version of the template defined in the ``django-oauth-toolkit`` library.

In an ideal world, this template would live in the ``oauth_dispatch`` djangoapp along with our other oauth2 code. Unfortunately, due to `the way that Django's app_directories loader works`_, we cannot put this template
in ``oauth_dispatch``, because the `template defined in django-oauth-toolkit`_ will take precedence over the one in ``oauth_dispatch``. The library must be defined first in ``INSTALLED_APPS``.
```